src/PrestaShopBundle/Form/Admin/Improve/International/Locations/CountryType.php

```diff
@@ -0,0 +1,189 @@
+                    new DefaultLanguage(),
+                ],
+            ])
+            ->add('iso_code', TextType::class, [
+                'required' => true,
+                'label' => $this->translator->trans('ISO code', [], 'Admin.International.Feature'),
+                'constraints' => [
+                    new Length([
+                        'max' => 2,
+                        'maxMessage' => $this->translator->trans(
+                            'This field cannot be longer than %limit% characters.',
+                            ['%limit%' => 2],
+                            'Admin.Notifications.Error'
+                        ),
+                    ]),
+                    new TypedRegex([
+                        'type' => TypedRegex::TYPE_LANGUAGE_ISO_CODE,
+                    ]),
+                ],
+            ])
+            ->add('call_prefix', TextType::class, [
+                'required' => true,
+                'label' => $this->translator->trans('Call prefix', [], 'Admin.International.Feature'),
+            ])
+            ->add('default_currency', ChoiceType::class, [
+                'required' => false,
+                'label' => $this->translator->trans('Default currency', [], 'Admin.International.Feature'),
+                'choices' => $this->currencyChoiceProvider->getChoices(),
+                'placeholder' => $this->translator->trans('Default store currency', [], 'Admin.International.Feature'),
+            ])
+            ->add('zone', ChoiceType::class, [
+                'required' => false,
+                'label' => $this->translator->trans('Zone', [], 'Admin.Global'),
+                'choices' => $this->zoneChoiceProvider->getChoices(
+                    [
+                        'active' => false,
+                        'active_first' => false,
+                    ]
+                ),
+                'placeholder' => false,
+            ])
+            ->add('need_zip_code', SwitchType::class, [
+                'required' => false,
+                'label' => $this->translator->trans('Does it need Zip/Postal code?', [], 'Admin.International.Feature'),
+            ])
+            ->add('zip_code_format', TextType::class, [
+                'required' => true,
+                'label' => $this->translator->trans(' Zip/Postal code format', [], 'Admin.International.Feature'),
+                'help' => $this->translator->trans('Indicate the format of the postal code: use L for a letter, N for a number, and C for the country\'s ISO 3166-1 alpha-2 code. For example, NNNNN for the United States, France, Poland and many other; LNNNNLLL for Argentina, etc. If you do not want PrestaShop to verify the postal code for this country, leave it blank.', [], 'Admin.International.Help'),
+                'constraints' => [
+                    new TypedRegex([
+                        'type' => TypedRegex::TYPE_ZIP_CODE_FORMAT,
+                    ]),
+                ],
```

Comment:
as mentioned in one of above comments, these 2 seems to be related. When "need zip code" is off, then zip_code_format should not be required. (probably could be done with form event listener, wdyt @jolelievre ?)

Reply:
well i can see that in old controller it didn't change the fact if format is required or not, i guess it ask to have format all the time but the first switch controls if customer needs to input it or if its not required. Not so sure how it works but isn't format always required that if customer inputs it we know if its correct format?

Reply:
I guess it's good enough for now, besides the JS component should handle the initial state